rend3-anim/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+    /// - instance: An instance of `scene`, as returned by
+    ///   [`load_gltf`](rend3_gltf::load_gltf) or
+    ///   [`instance_loaded_scene`](rend3_gltf::instance_loaded_scene)
+    pub fn from_gltf_scene(scene: &LoadedGltfScene, instance: &GltfSceneInstance) -> Self {
+        // The set of joints that each animation affects, stored as node indices
+        let animation_to_joint_nodes: HashMap<AnimationIndex, HashSet<NodeIndex>> = scene
+            .animations
+            .iter()
+            .enumerate()
+            .flat_map(|(anim_idx, anim)| {
+                anim.inner
+                    .channels
+                    .keys()
+                    .map(move |node_idx| (AnimationIndex(anim_idx), NodeIndex(*node_idx)))
+            })
+            .into_grouping_map()
+            .collect::<HashSet<_>>();
+
+        let mut animation_skin_usage = HashMap::<AnimationIndex, Vec<SkinIndex>>::new();
+        for animation_idx in 0..scene.animations.len() {
+            let animation_idx = AnimationIndex(animation_idx);
+            for (skin_index, skin) in scene.skins.iter().enumerate() {
+                let skin_index = SkinIndex(skin_index);
+
+                let anim_affected_nodes = &animation_to_joint_nodes[&animation_idx];
+                if skin
+                    .inner
+                    .joints
+                    .iter()
+                    .any(|j| anim_affected_nodes.contains(&NodeIndex(j.inner.node_idx)))
+                {
+                    let entry = animation_skin_usage
+                        .entry(animation_idx)
+                        .or_insert_with(Default::default);
+                    entry.push(skin_index);
+                }
+            }
+        }
+
+        let mut skin_data = HashMap::new();
+        for (skin_index, skin) in scene.skins.iter().enumerate() {
+            let skin_index = SkinIndex(skin_index);
+
+            let node_to_joint_idx = skin
+                .inner
+                .joints
+                .iter()
+                .enumerate()
+                .map(|(idx, joint)| (NodeIndex(joint.inner.node_idx), JointIndex(idx)))
+                .collect();
+
+            // Nodes affected by this skin (i.e. joints)
+            let skin_nodes: Vec<NodeIndex> = skin.inner.joints.iter().map(|j| NodeIndex(j.inner.node_idx)).collect();
+
+            let joint_nodes_topological_order: Vec<NodeIndex> = instance
+                .topological_order
+                .iter()
+                .map(|node_idx| NodeIndex(*node_idx))
+                .filter(|node_idx| skin_nodes.contains(node_idx))
+                .collect();
+
+            let skeletons: Vec<SkeletonHandle> = instance
+                .nodes
+                .iter()
+                .flat_map(|node| &node.inner.object)
+                .flat_map(|object| &object.inner.armature)
+                .filter(|armature| armature.skin_index == skin_index.0)
+                .flat_map(|armature| &armature.skeletons)
```

Comment:
I picture you with a big hammer just hitting this code saying "flatter! flatter!" and it amuses me